### PR TITLE
Update inefficient queries.

### PR DESCRIPTION
**Pull request by jmcarp**
Use `select_related` and `prefetch_related` as appropriate to avoid n+1
queries in a handful of views. Use the `nplusone` query profiler in test
mode to detect similar inefficient queries in the future.

Note: I'm the maintainer of `nplusone`--would be interested in any feedback folks might have about the library, bug reports, etc.

Further reading: http://use-the-index-luke.com/sql/join/nested-loops-join-n1-problem

### CODE REVIEW

**Comment by ccostino** [comment]
Big +1 on this.  /cc @batemapf

Thanks for sharing `nplusone` too, @jmcarp, that is awesome!  It would have been extraordinarily helpful for a Django app I was working on several years ago.  I will be sure to keep it in mind for the future, especially if I find myself doing report generation across several models again. :-)

**Comment by jmcarp** [comment]
I saw that a few minor merge conflicts came up after I submitted this--they're resolved now.
cc @ccostino @batemapf

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,6 +1 @@
-django-debug-toolbar
-django-webtest
-factory-boy
-coverage
-ipython
-ipdb
```
Comment (ccostino) [inline requirements-dev.txt]: I'm not opposed to leaving IPython and ipdb out of the dev dependencies (though I find them helpful!), I'm just curious if you had a specific reason for omitting them.

File: tock/hours/forms.py [hunk 2]
```diff
@@ -136,7 +137,7 @@ def clean_project(self):
         return data
 
     def clean(self):
-        cleaned_data = super(TimecardObjectForm, self).clean()
+        super(TimecardObjectForm, self).clean()
```
Comment (ccostino) [inline tock/hours/forms.py]: D'oh!  Should've caught that with my previous PR, thanks!

**Comment by ccostino** [comment]
Awesome, thank you @jmcarp!  I just took a look and everything appears fine to me, just a couple of notes.  @batemapf, this would be great to have in place!

**Pull request merged**